source/h2_stream.c

```diff
@@ -358,6 +415,73 @@ static void s_stream_update_window(struct aws_http_stream *stream_base, size_t i
     }
 }
 
+static int s_stream_reset_stream(struct aws_http_stream *stream_base, enum aws_http2_error_code http2_error) {
+
+    struct aws_h2_stream *stream = AWS_CONTAINER_OF(stream_base, struct aws_h2_stream, base);
+    struct aws_h2_connection *connection = s_get_h2_connection(stream);
+    int err = 0;
+    bool stream_is_activated;
+    bool cross_thread_work_should_schedule = false;
+
+    { /* BEGIN CRITICAL SECTION */
+        s_lock_synced_data(stream);
+
+        err |= stream->synced_data.requested_reset_error_code != AWS_HTTP2_ERR_COUNT;
+        stream_is_activated = stream->synced_data.is_activated;
+        if (!err && stream_is_activated) {
+            cross_thread_work_should_schedule = !stream->synced_data.is_cross_thread_work_task_scheduled;
```

Comment:
We probably ALSO shouldn't send RST_STREAM if the stream is complete?

The spec says you shouldn't send multiple RST_STREAM
But really it probably just opens us up to a bunch of weird bugs to let users queue work after we know the stream is closed